src/Autofac/Core/Registration/ModuleRegistrar.cs

```diff
@@ -12,10 +12,7 @@ namespace Autofac.Core.Registration
     /// </summary>
     internal class ModuleRegistrar : IModuleRegistrar
     {
-        /// <summary>
-        /// The <see cref="ContainerBuilder"/> into which registrations will be made.
-        /// </summary>
-        private readonly ContainerBuilder _builder;
+        private Action<IComponentRegistryBuilder>? _nextModuleCallback;
```

Comment:
It took me a bit to figure out that this is effectively the call to `module.Configure` for all modules registered via this registrar. The `RegistrationData` allows wrapping of that chain with predicates, but `RegisterModule` needs the ability to add modules _inside that wrapper_.

I wonder if a more descriptive name like `_moduleConfigureChain` or something might help? I'm not glued to that as the name, but it did take me a few reads through here to go, "OOOHHH, it's predicates wrapped around this callback, and this callback gets sort of lazy-initialized via closure." Like two levels of callbacks, not just one "pipeline." I think `_nextModule` threw me off because I was expecting one pipeline, like ASP.NET Core `_next` and my poor zombie brain didn't connect that it's sorta two levels going on here.

I also had a big block of `//` comments written but was like... maybe it's just me and a more descriptive name would be enough.

Reply:
The word 'next' does imply pipeline, you're right (in Autofac as well, never mind ASP.NET Core). Will re-word and add some comments.